sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -408,6 +421,23 @@ object JdbcUtils extends Logging {
       (rs: ResultSet, row: InternalRow, pos: Int) =>
         row.setFloat(pos, rs.getFloat(pos + 1))
 
+
+    // SPARK-33888 - sql TIME type represents as physical int in millis
+    // Represents a time of day, with no reference to a particular calendar,
+    // time zone or date, with a precision of one millisecond.
```

Comment:
After a second thought, why do we pick millisecond precision? Why not microsecond? Is there a standard for it?

Reply:
Since we are converting to/from java.sql.Time, and according to the javadoc https://docs.oracle.com/javase/8/docs/api/java/sql/Time.html , it supports till milliseconds for constructor.

Reply:
It may confuse Spark users, as Spark timestamp is microsecond precision.

After more thought, it's probably better to return timestamp when reading JDBC time, with a clear rule: we convert the time to timestamp by using "zero epoch" as the date part. It's also more useful as users can call `hour` function or similar ones to get some field values. What do you think?

Reply:
Hmm, I agree with you on the user experience part (`hour` function and all). I think it is hard (near impossible) to introduce new DataType (Time - HH:MM:SS.sss display) and another function to convert an int without date portion to Timestamp (most conversion is parse string) as it gonna take through multiple level of approvals and testings.

Another reason why I went with Int was that with TimestampType, it breaks compatibility with Avro logical & Spark type converter (https://github.com/apache/spark/blob/master/external/avro/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala).

I'm not sure if a `time_to_timestamp()` helper function would be a better compromise or revert back the 2 MRs?

Reply:
It looks better if the avro schema converter can convert timestamp to time. After reading time column from JDBC, it becomes `IntegerType` and there is no context to indicate that this int comes from JDBC time and means milliseconds. What if the avro logic type is time-micros? With timestamp type, at least we know the precision is microsecond.

Reply:
Your suggestion makes sense. We can also stuff the info into the metadata field of the struct field. Let me find sometimes this weekend or over the new year to try out your suggestion? Will ping you back if I couldn't find time or manage to get a solution so we can revert the MRs. Thanks a lots for helping me out!

Should I create a new JIRA ticket and cut a new MR or do you prefer to consolidate in here?

Reply:
Let's create a new JIRA to track it. The PR is merged to master only so we have plenty of time to fix it before the 3.2 release :)

Reply:
@cloud-fan btw, I did a quick check and it seems like if we use TimestampType, the time will always be converted to the JVM system timezone. So for the JDBCSuite test, given `12:34:56` time value, when you do the `.select(hour("time"))` it will always point to my local timezone hour instead of `12`. So I don't know if we should proceed in this case.

Reply:
We can do whatever we want. We can use JDBC API `getTime` to get the time value, and construct the timestamp value in a reasonable way. It's under our control.